prime-router/src/main/kotlin/Metadata.kt

```diff
@@ -357,23 +439,32 @@ class Metadata {
         const val schemaExtension = ".schema"
         const val valueSetExtension = ".valuesets"
         const val tableExtension = ".csv"
-        const val defaultMetadataDirectory = "./metadata"
+        private const val defaultMetadataDirectory = "./metadata"
         const val schemasSubdirectory = "schemas"
         const val valuesetsSubdirectory = "valuesets"
         const val tableSubdirectory = "tables"
         const val fileNameTemplatesSubdirectory = "./file_name_templates"
-        @Volatile private var defaultMetadata: Metadata? = null
-        // I am probably threadsafe. If things go bananas verify I'm not the cause
-        // this is instead of doing everything with DI because doing DI right in Azure
-        // with existing DI libraries was extremely complex and beyond the scope
-        // of this work. And probably hard to get right. And honestly not necessary. Probably.
-        // honestly, the case could be made to make Metadata a singleton and then this code
-        // can go away, but that is also beyond the scope of this work right now
-        fun provideMetadata(): Metadata {
-            return defaultMetadata ?: synchronized(this) {
-                val newInstance = defaultMetadata ?: Metadata(defaultMetadataDirectory).also { defaultMetadata = it }
-                newInstance
-            }
+
+        /**
+         * Singleton object
+         */
+        private val singletonInstance: Metadata by lazy(LazyThreadSafetyMode.SYNCHRONIZED) {
+            Metadata(defaultMetadataDirectory)
         }
+
+        /**
+         * Get the singleton instance.
+         */
+        fun getInstance(): Metadata {
+            // Load any updates to the database lookup tables.
+            // Due to the use of this function, this will get checked at least on every Azure function call
+            singletonInstance.checkForDatabaseLookupTableUpdates()
+            return singletonInstance
+        }
+
+        /**
+         * The amount of seconds to wait before tables are checked again for updates.
+         */
+        private val tablePollInternalSecs = if (Environment.get() == Environment.PROD) 300 else 30
```

Comment:
I've had plans for months to expand the settings table or add a new table to enable per-environment settings such as these.  (Or, it occurs to me that we could use the LookupTable to store per-environment variables.  ;) )

Reply:
We can do that, just need to plan it good.  Some of the values will not change, though.

Reply:
I like this.  I will need to find out how we tell DevOps how to set that up